integ-tests/src/test/java/com/google/integration/IntegrationTests.java

```diff
@@ -80,5 +86,31 @@ public void usersTest() throws SQLException, IOException {
           AvroHelper.getLongNotNull(record, "LastAccessTimeStamp");
       avroList.add(avroRow);
     }
+
+    List<UserTableRow> dbListCopy = new ArrayList<>(dbList);
```

Comment:
I'm just curious about this pattern. Usually, tests use the pattern where the assert something and provide an error message if the assertion fails. in [ExtractExecutorImplTest line 73](For example https://github.com/google/dwh-assessment-extraction-tool/blob/main/src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImplTest.java#L73) we assert that the outcome is 0.

Here you rather have an if check, log separately (i.e., not use the underlying unit test framework) and then fail manually.

I think the whole if/else ladder here could be replaced with

```
assertThat(dbList).isEqualTo(avroList);
```

IIRC, the truth library's [`IterableSubject#isEqualTo`](https://truth.dev/api/latest/com/google/common/truth/IterableSubject#isEqualTo(java.lang.Object)) provides a diff if the equals test fails.

In short, what is the purpose of the manual logging instead of letting the unit testing framework take care of that?

Reply:
I agree, this if-else ladder is bulky, but it gives a clear answer what went wrong printing out the proper diff.

Ok, I tried this 'assertThat(dbList).isEqualTo(avroList);' , I personally barely may understand what is missing or where there is an extra row. I can show you it on a call.

Reply:
Done.